mysql-plugin/src/main/java/io/cdap/plugin/mysql/MysqlConnectorConfig.java

```diff
@@ -58,6 +60,8 @@ public Properties getConnectionArgumentsProperties() {
     prop.put(JDBC_PROPERTY_CONNECT_TIMEOUT, "20000");
     prop.put(JDBC_PROPERTY_SOCKET_TIMEOUT, "20000");
     prop.put(JDBC_REWRITE_BATCHED_STATEMENTS, "true");
+    // MySQL property to ensure that TINYINT(1) type data is not converted to MySQL Bit/Boolean type in the ResultSet.
+    prop.put(MYSQL_TINYINT1_IS_BIT, "false");
```

Comment:
This is a backward incompatible change right?

Reply:
On a second thought, we already support adding connection arguments whenever needed why do we need to change the default behaviour?

Reply:
Without this connection argument, for Tinyint(1) type MySQL driver code internally converts these type into boolean type, Due to which any value between [-128, 127] becomes true/false(for 0).

Reply:
But this is the issue with mysql driver not the plugin what if the existing pipelines are depending on the boolean conversion?

Any customer want to change that behaviour they can always set `MYSQL_TINYINT1_IS_BIT, "false"` in connection arguments.

@albertshau @tivv wdyt??

Reply:
what was the previous behavior? Would we detect a tinyint(1) column as a CDAP int or boolean?

If we detected it as an CDAP int, then I imagine we were setting a boolean for an int field in the StructuredRecord? If so that was a bug and we can just do this fix without worrying about compatibility.

If we detected it and read it as a CDAP boolean, then this is a backwards incompatible change and shouldn't be done.

Reply:
The previous behavior was that tinyint(1) gets detected as CDAP boolean. 
This change can be done in the backward compatible way. 
@albertshau Should the default behavior be that tinyint(1) be detected as int instead of a bool ?

Reply:
I think it makes more sense for it to be an int by default, since there is a separate boolean type.